File: solutions/fsharp/log-levels/2/LogLevels.fs
module LogLevels

open Logging.Level
open Logging.LogUtilities

let message (logLine: string): string =
    handleLogEntry logLine (fun logEntry -> logEntry.Message)

let logLevel (logLine: string): string =
    handleLogEntry logLine (fun logEntry -> levelToString logEntry.Level)

let reformat (logLine: string): string =
    handleLogEntry logLine (fun logEntry -> $"%s{logEntry.Message} (%s{levelToString logEntry.Level})")

File: solutions/java/log-levels/2/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
		return new Log(logLine)
				       .getMessage();
    }

    public static String logLevel(String logLine) {
	    return new Log(logLine)
			           .getLevel();
    }

    public static String reformat(String logLine) {
	    return new Log(logLine)
			           .toString();
    }
}
